### ISSUE TITLE
Broken scenario or some logic?

### ISSUE THREAD

**Issue by SirYaro** [Tarmslitaren/FrosthavenAssistant#12]
Hi 
We just noticed some kind of issue with scenario 12 for JotL. 

If you turn it on for the first time, it will load a lot of "Pillars monsters" (6 if I recall correctly). Almost all was good until we tried to reload the scenario. (We chose a different one and then 12th again). We quickly realized that pillars won't load up and there is no way to add them back manually, or in any other way.

Almost nothing is able to bring the pillars back. I tried multiple combination that included exiting, reloading scenario, removing monsters, characters, changing campaign and reloading scenario again. In 30+ tries, it worked only once. So I have no idea what is causing this issue.

You remember I wrote "Almost all was good"? :) There might be something else on top of that.
I'm not 100% sure (because I wasn't able to get Pillars back to double-check that) but I think changing scenario level doesn't change pillar's hit points, and/or they might use the wrong formula for HP calculation. On LVL 3 scenario, when I load it for the first time, Pillars had 6 and (I think) 8HP instead of 8 and 12. But I'm there might be something else going on here. I wasn't mentally prepare for debugging so this report is not very accurate.

I got the same issue on android tablet, and on VM running windows 10.

**Comment by Tarmslitaren** [comment]
The first issue (objects/escorts not loading up the second time a scenario is set ) is definitely a bug. I will fix it for next version.  You can get around it by  killing the app and restarting. There is also an option to manually add escorts/objects from the add character menu.

The second issue (objects health not updating with the scenario level) is a known issue. It works the same way as the old Gloomhaven Helper so I've lazily not bothered to fix it. Will look into this as well for next version. In the meantime: Please set the desired scenario level before setting a scenario, and the objectives and escorts will have the correct health. 

The same goes for scenarios where some monsters have a different monster level than scenario level due to special rules: they will need to be manually set correctly after changing the scenario level.

Thanks for reporting!

**Comment by SirYaro** [comment]
Hey Hans
Thanks for getting back to me.
>  I will fix it for next version

Danke :)
>  You can get around it by killing the app and restarting.

I'm not sure about that. App have some kind of state cache, so If I close or kill app it will start exactly in the same state as it was before. Which is GREAT in any situation except this :)
> There is also an option to manually add escorts/objects from the add character menu.

That is great! I did not notice that. I was looking for something like this in "add monster" menu since they are "enemies" not "characters". Perhaps you could consider moving it, if there is no particular reason to keep it there? Or make it available in both places?
> Please set the desired scenario level before setting a scenario, and the objectives and escorts will have the correct health.

There is something wrong with this, I think.
I add two 5lvl chars, add scenario 12, manually change lvl to 3 (from 3 tbh), then:
- add character / objective / lvl 3
- add character / objective / lvl 2
- add character / objective / lvl 1
What I'm getting are objectives with 14, 10 and 6 HP, which I believe is wrong. In our scenario booklet, pillars HP are described with TWO formulas: 
1) HP = (SL +1) x PC, and
2) HP = (SL +3) x PC, where:
SL = scenario lvl
PC = player count
So, with 2 previously mentioned players, I should get:
ad 1) 8, 6, 4 HP
ad 2) 12, 10, 8 HP
As you see, none of them matches received values.
Maybe instead of choosing objective lvl it would be better to set HP max manually instead?

------

I also have 3 unrelated queries if you don't mind :) If you prefer separate issues raised for each of them, just let me know. I'll change that.
1) what's the android app requirements? Whatever I tried, I could not install it on virtual tablet running on VM (created with genymotion). In google shop app is not visible on such device, and after I found apk available here I could not install it neither...
2) Is there any way to add a card to the monster modifier deck? If they got a blessing or a curse, there is no way to reflect that.
3) improvement suggestion. Right now, scrolling up/down works only on a character tab width and not on marked areas:
![Screenshot_20221029-124411](https://user-images.githubusercontent.com/5879723/198827220-3c033e39-c17b-472d-a98c-d81f546eda4c.jpg)
Can you extend it?

**Comment by Tarmslitaren** [comment]
Ok, lets see if I can answer all:
1.  there is a state cache, and it will auto save after every action. Though I've yet to figure out the bug, I think just setting the scenario again should fix it, and if not then first change scenario, then kill app and then set the scenario you want.
2. I will consider other menu for adding escorts/objectives, but most of the time it's not something you would be interested in doing manually anyway. Same with monsters really. Unless you are playing a custom scenario there is no reason to add monsters or objectives, instead of just setting a scenario.
3. The 'level' of objectives/escorts is a bit arbitrary, admittedly, but you can set a custom max health from the same menu you set the character level on. It's available also from the status menu - press the crown
4 (1). Android version support should be 4.1 and higher. I'm not familiar with genymotion VM settings, sorry.
5 (2). curses and blesses can be added from 2 places: A: from modifier deck menu accessible by tapping the 'discard' pile. and B: from a monsters status menu (opened by tapping a monster standee box).
6 (3). scrolling from whole view: I'd rather not touch this code more. It will be hard to work around in the case you have more than one column that will scroll independently. (you may change main list size in settings, and if it covers less than half the width of the screen, you will get 2 columns)

**Comment by SirYaro** [comment]
> I think just setting the scenario again should fix it, 

It doesn't. I can record video 4 you if you have problems replicating this issue.
> and if not then first change scenario, then kill app and then set the scenario you want

It works that way. Thanks a lot.

---
My son just remind me of one feature we think would be nice. I'm not sure how this status
![bleed](https://user-images.githubusercontent.com/5879723/198837958-e2673bda-697a-4fa4-9daf-4723ca3cbe8a.jpg)
is called in English, but it causes target to loose 1HP every turn. Could this be automated?

**Comment by Tarmslitaren** [comment]
Automatically apply wound? I suppose I can put it on my backlog, but it's low priority and may not be done in quite a while.

**Comment by SirYaro** [comment]
> Automatically apply wound? I suppose I can put it on my backlog, but it's low priority and may not be done in quite a while.

No worries, it's not urgent. :) It just looks like a neat addition in terms of boring and easy to forget stuff automation. You already implemented a "shaking wound icon" so I thought it wouldn't be too difficult to add one more function right after that :)
